### PR TITLE
WIP: Some changes

### PR DESCRIPTION
**Pull request by woylie**
I've made some changes here and there. Still a lot to do, but please have a look.

### CODE REVIEW

File: lib/blue_bird/controller.ex [hunk 1]
```diff
@@ -1,50 +1,89 @@
...
+  Generates the description of a route.
+
+  - `method`: HTTP method (GET, POST, PUT etc.)
+  - `url`: Route as defined in the Phoenix router
+  - `group`: Resource group. Defaults to resource name guessed from controller
+  name.
+  - `resource`: Title for the resource. Defaults to the url.
+  - `title`: Title for the action
+  - `description`: Description of the route
+  - `note`: Note
```
Comment (rhazdon) [inline lib/blue_bird/controller.ex]: The parameter `warning` is missing here.

File: lib/blue_bird/controller.ex [hunk 2]
```diff
@@ -1,50 +1,89 @@
...
+        group "Posts"
+        resource "Single Post"
+        title "Show post"
+        description "Show post by ID"
+        parameter :id, :integer, :required, "User ID"
+        parameter :id, :integer, "Post ID"
+        parameter :name, :string
+      end
   """
+  # TODO: If parameter is not required and not set, it will be another route.
```
Comment (rhazdon) [inline lib/blue_bird/controller.ex]: You are absolutely right. In fact, this parameter is for the defined route always required. Therefore you can actually omit this. I also think about omitting this `parameter` option completely and tp read and determine these parameters from the `conn.path_params`. 

What do you think about it?

File: lib/blue_bird/controller.ex [hunk 3]
```diff
@@ -53,36 +92,57 @@ defmodule BlueBird.Controller do
...
-      end
+    cond do
+      is_nil(values) -> nil
+      length(values) == 1 -> List.first(values)
+      true -> raise ArgumentError,
+              "Expected single value for #{key}, got #{length(values)}"
     end
   end
 
-  defp atom_to_string(atom_or_string) do
```
Comment (rhazdon) [inline lib/blue_bird/controller.ex]: Very welcome, thank! :D

File: lib/blue_bird/formatter.ex [hunk 4]
```diff
@@ -1,37 +1,33 @@
...
   @doc """
-  Listen to events.
-
-  If the event is `:suite_finished`, trigger the generation of api blueprint file.
-
-  Ignore  all other events.
+  Event listener. Triggers the generation of the api blueprint file on
+  `:suite_finished`. Ignores all other events.
   """
   def handle_event({:suite_finished, _run_us, _load_us}, nil) do
```
Comment (rhazdon) [inline lib/blue_bird/formatter.ex]: Obviously... xD

**Comment by rhazdon** [comment]
Wow! Thank you very much! Great work. :)

File: lib/blue_bird/controller.ex (hunk 2 above)
Comment (woylie) [inline lib/blue_bird/controller.ex]: We still need to be able to add descriptions to the parameters, so we need the parameter option. Instead I would like to emit warnings or raise errors if the the specified parameters don't match the route.

File: lib/blue_bird/formatter.ex (hunk 4 above)
Comment (woylie) [inline lib/blue_bird/formatter.ex]: We don't see the code in the documentation ;)

**Comment by woylie** [comment]
Still work in progress. Please don't merge yet. 🙃

**Pull request merged**